sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -1027,13 +1027,13 @@ protected void prepareWrite(WritableByteChannel channel) throws Exception {
       }
 
       @Override
-      public void write(T value) throws Exception {
-        dataFileWriter.append(value);
+      protected void finishWrite() throws Exception {
+        dataFileWriter.flush();
       }
 
       @Override
-      protected void writeFooter() throws Exception {
-        dataFileWriter.flush();
+      public void write(T value) throws Exception {
+        dataFileWriter.append(value);
```

Comment:
This is an _optional_ "nit", but:

Swap the order back? I think it makes sense to list the functions in "logical" order -- prepare, write, finishWrite. That way it kind of reads top to bottom in the right way.